contracts/token/ERC20/utils/SafeERC20.sol

```diff
@@ -1,9 +1,9 @@
 // SPDX-License-Identifier: MIT
 // OpenZeppelin Contracts (last updated v4.8.0) (token/ERC20/utils/SafeERC20.sol)
 
-pragma solidity ^0.8.1;
+pragma solidity ^0.8.18;
 
-import "../IERC20.sol";
+import "../ERC20.sol";
```

Comment:
If we could avoid that it would be better. Devs are going to use this library in contracts that are not ERC20. Importing ERC20 would carry a lot of code that is not needed. 

I'm afraid this will not only reduce the readability of the verified code, but also confuse the feature detection that is based on the verified source code.

@frangio what do you think ?

Reply:
I don't think this affects feature detection. But it does bring in some code and may impact compilation times.

Reply:
Isn't there a thing were we look at the verified source code for a contract, and if we find that "ERC20.sol" is part of the verified files we flag that as "contract uses ERC20" ?

Reply:
it could emit ERC20 errors without loading ERC20.sol if they were declared in an `library ERC20Errors` using the syntax.

Reply:
They are independent decisions and one depends on the other. So, should SafeERC20 emit ERC20 errors, or its own kind of error?

Reply:
These are currently [my conclusions](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4261#issuecomment-1569320526) on the idea of library errors. However, we haven't solved the issue of importing `ERC20.sol` yet, so I'm leaning towards @frangio proposal of having `ERC20ExceededAllowanceDecrease` renamed to `SafeERC20ExceededAllowanceDecrease` on its own domain.

Do you guys agree?

Reply:
I move forward and replaced with `SafeERC20ExceededAllowance` since I think it's the most agreed solution but let me know otherwise.